v1/plugin/plugin.go

```diff
@@ -302,8 +368,137 @@ type preamble struct {
+		}
+		pluginProxy = &proxy.pluginProxy
+		server, meta, err = buildGRPCServer(processorType, appArgs.name, appArgs.version, arg, appArgs.opts...)
+		if err != nil {
+			panic(err)
+		}
+		rpc.RegisterProcessorServer(server, proxy)
+	case Publisher:
+		proxy := &publisherProxy{
+			plugin:      plugin,
+			pluginProxy: *newPluginProxy(plugin),
+		}
+		pluginProxy = &proxy.pluginProxy
+		server, meta, err = buildGRPCServer(publisherType, appArgs.name, appArgs.version, arg, appArgs.opts...)
+		if err != nil {
+			panic(err)
+		}
+		rpc.RegisterPublisherServer(server, proxy)
+	case StreamCollector:
+		proxy := &StreamProxy{
+			plugin:             plugin,
+			pluginProxy:        *newPluginProxy(plugin),
+			maxCollectDuration: defaultMaxCollectDuration,
+			maxMetricsBuffer:   defaultMaxMetricsBuffer,
+		}
+		pluginProxy = &proxy.pluginProxy
+		server, meta, err = buildGRPCServer(publisherType, appArgs.name, appArgs.version, arg, appArgs.opts...)
+		if err != nil {
+			panic(err)
+		}
+		rpc.RegisterStreamCollectorServer(server, proxy)
+	default:
+		logger.WithField("type", fmt.Sprintf("%T", plugin)).Fatal("Unknown plugin type")
+	}
+
+	if c.Bool("stand-alone") {
+		httpPort := c.Int("stand-alone-port")
+		preamble, err := printPreambleAndServe(server, meta, pluginProxy, arg.ListenPort, arg.Pprof)
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+
+		go func() {
+			http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+				fmt.Fprintln(w, preamble)
+			})
+
+			listener, err := net.Listen("tcp", fmt.Sprintf(":%d", httpPort))
+			if err != nil {
+				panic("Unable to get open port")
+			}
+			defer listener.Close()
+			fmt.Printf("Preamble URL: %v\n", listener.Addr().String())
+			err = http.Serve(listener, nil)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}()
+		<-pluginProxy.halt
+
+	} else if libInputOutput.args() > 0 {
+		// snapteld is starting the plugin
+		// presumably with a single arg (valid json)
+		preamble, err := printPreambleAndServe(server, meta, pluginProxy, arg.ListenPort, arg.Pprof)
+		if err != nil {
+			log.Fatal(err)
+		}
+		libInputOutput.printOut(preamble)
+		go pluginProxy.HeartbeatWatch()
+		<-pluginProxy.halt
+
+	} else {
+		// no arguments provided - run and display diagnostics to the user
+		var config Config
+		if c.IsSet("config") {
+			err := json.Unmarshal([]byte(c.String("config")), &config)
+			if err != nil {
+				return fmt.Errorf("! Error when parsing config. Please ensure your config is valid. \n %v", err)
+			}
+		}
+
+		switch pluginProxy.plugin.(type) {
+		case Collector:
+			showDiagnostics(*meta, pluginProxy, config)
+		case Processor:
+			fmt.Println("Diagnostics not currently available for processor plugins.")
+		case Publisher:
+			fmt.Println("Diagnostics not currently available for publisher plugins.")
+		}
+	}
+	return nil
+}
+
+func printPreambleAndServe(srv server, m *meta, p *pluginProxy, port string, isPprof bool) (string, error) {
```

Comment:
I suppose that by setting `--port` and `--isPprof` the values of input args (port and isPprof) will change, but I cannot achieve that. Below is what I did:
- I added debug logs:
![image](https://cloud.githubusercontent.com/assets/11335874/25350393/4dccddc6-2925-11e7-9216-728ae4bad8f4.png)

- I run the following command: ./snap-plugin-collector-rand --log-level 5 --port 8184 --pprof --stand-alone

    - what I received:
    ```
    DEBU[0000] Debug port is=
    DEBU[0000] Debug isPprof=false
    ```

    - what I expected is port equals 8184 and isPprof set to true